File: CS/WpfApplication1/SampleGalleryViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WpfApplication1 {
    public class SampleGalleryViewModel : ViewModelBase {
        public ObservableCollection<GroupViewModel> Groups {
            get { return GetProperty(() => Groups); }
            set { SetProperty(() => Groups, value); }
        }
        public SampleGalleryViewModel() {
            Groups = new ObservableCollection<GroupViewModel>();
            InitGroups();
        }
        void InitGroups() {
            for(int i = 0; i < 10; i++) {
                Groups.Add(new GroupViewModel() { Caption = String.Format("Group {0}", i) });
            }
        }
    }
    public class GroupViewModel : ViewModelBase {
        public string Caption {
            get { return GetProperty(() => Caption); }
            set { SetProperty(() => Caption, value); }
        }
        public ObservableCollection<ItemViewModel> Items {
            get { return GetProperty(() => Items); }
            set { SetProperty(() => Items, value); }
        }
        public GroupViewModel() {
            Items = new ObservableCollection<ItemViewModel>();
            InitItems();
        }
        void InitItems() {
            for(int i = 0; i < 3; i++) {
                Items.Add(new ItemViewModel() { Caption = String.Format("Item {0}", i) });
            }
        }
    }
    public class ItemViewModel : ViewModelBase {
        public string Caption {
            get { return GetProperty(() => Caption); }
            set { SetProperty(() => Caption, value); }
        }
    }
}

File: VB/WpfApplication1/SampleGalleryViewModel.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq
Imports System.Linq.Expressions
Imports System.Text
Imports System.Threading.Tasks

Namespace WpfApplication1
	Public Class SampleGalleryViewModel
        Public Overridable Property Groups() As ObservableCollection(Of GroupViewModel)
        Public Sub New()
			Groups = New ObservableCollection(Of GroupViewModel)()
			InitGroups()
		End Sub
		Private Sub InitGroups()
			For i As Integer = 0 To 9
				Groups.Add(New GroupViewModel() With {.Caption = String.Format("Group {0}", i)})
			Next i
		End Sub
	End Class
	Public Class GroupViewModel
		Inherits ViewModelBase
        Public Overridable Property Caption() As String	
        Public Overridable Property Items() As ObservableCollection(Of ItemViewModel)
		Public Sub New()
			Items = New ObservableCollection(Of ItemViewModel)()
			InitItems()
		End Sub
		Private Sub InitItems()
			For i As Integer = 0 To 2
				Items.Add(New ItemViewModel() With {.Caption = String.Format("Item {0}", i)})
			Next i
		End Sub
	End Class
	Public Class ItemViewModel
        Public Overridable Property Caption() As String
	End Class
End Namespace
